### PR TITLE
Type definitions for node's net module

### PR DESCRIPTION
**Pull request by JohnyDays**

### CODE REVIEW

**Comment by JohnyDays** [comment]
Question: Is there a way to event emitter listener argument values based on the event?

File: lib/node.js [hunk 1]
```diff
@@ -664,9 +664,47 @@ declare module "https" {
...
+
   declare function isIP(input: string): number;
   declare function isIPv4(input: string): boolean;
   declare function isIPv6(input: string): boolean;
+
+
+  declare type connectionListener = (socket: Socket) => any;
+  declare function createServer(
+    options?: {
+      allowHalfOpen: boolean,
```
Comment (mroch) [inline lib/node.js]: i think these keys should be optional too. `net.createServer({pauseOnConnect: false})` is valid, right?

Comment (JohnyDays) [inline lib/node.js]: Makes perfect sense, I'll add that

**Comment by mroch** [comment]
@facebook-github-bot shipit

**Pull request closed without merging**